### PR TITLE
Changes for Shooting and Shot events

### PR DESCRIPTION
**Pull request by ghost**
- Added new properties in `ShotEventArgs` : `Firearm` and `Item` (+ it now extends IFirearmEvent)
- Fixed Shot event patch for Buckshot and Particle Disruptor
- `ShootingEventArgs` now extends IFirearmEvent instead of IItemEvent
- Edited `Shot.ProcessShot` method documentation

### CODE REVIEW

File: Exiled.Events/Patches/Events/Player/Shot.cs [hunk 1]
```diff
@@ -35,16 +35,17 @@ namespace Exiled.Events.Patches.Events.Player
...
+        /// <param name="player">The player.</param>
+        /// <param name="firearm">The firearm.</param>
+        /// <param name="hit">The raycast hit.</param>
+        /// <param name="destructible">The destructible.</param>
+        /// <param name="damage">The damage.</param>
+        /// <returns>If the shot is allowed.</returns>
+        internal static bool ProcessShot(ReferenceHub player, Firearm firearm, RaycastHit hit, IDestructible destructible, ref float damage)
         {
-            ShotEventArgs shotEvent = new(Player.Get(player), hit, destructible, damage);
+            ShotEventArgs shotEvent = new(Player.Get(player), (API.Features.Items.Firearm)API.Features.Items.Item.Get(firearm), hit, destructible, damage);
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Player/Shot.cs:48]: Use `Cast<Firearm>()` instead.

**Review by NaoUnderscore — CHANGES_REQUESTED** [review]

File: Exiled.Events/Patches/Events/Player/Shot.cs (hunk 1 above)
Comment (SiphoxR) [reply Exiled.Events/Patches/Events/Player/Shot.cs]: Done

**Review by louis1706 — CHANGES_REQUESTED** [review]
Add IFirearmEvent at the eventargs class

**Comment by SiphoxR** [comment]
I'll add it when I'm at home

**Comment by SiphoxR** [comment]
Btw I realised that Shooting event has a firearm property too but doesn't implement the interface

**Comment by louis1706** [comment]
> Btw I realised that Shooting event has a firearm property too but doesn't implement the interface

You can add it so ^^

**Comment by SiphoxR** [comment]
In this pr or another one?

File: Exiled.Events/Patches/Events/Player/Shot.cs [hunk 2]
```diff
@@ -35,16 +37,17 @@ namespace Exiled.Events.Patches.Events.Player
     internal static class Shot
     {
         /// <summary>
-        /// I DON'T CARE.
+        /// Process shot.
```
Comment (louis1706) [inline Exiled.Events/Patches/Events/Player/Shot.cs:40]: Ahahahh nice one lol

**Review by louis1706 — APPROVED** [review]
Everything seems good

**Review by NaoUnderscore — APPROVED** [review]

**Pull request merged**